lib/matplotlib/cbook/__init__.py

```diff
@@ -1711,11 +1714,15 @@ def normalize_kwargs(kw, alias_mapping=None, required=(), forbidden=(),
     TypeError
         To match what python raises if invalid args/kwargs are passed to
         a callable.
-
     """
+    from matplotlib.artist import Artist
+
     # deal with default value of alias_mapping
     if alias_mapping is None:
         alias_mapping = dict()
+    elif (isinstance(alias_mapping, type) and issubclass(alias_mapping, Artist)
```

Comment:
Why do you require subclasses of `Artist`? Wouldn't it be enough to use ducktyping via `getattr(alias_mapping, "_alias_map", {})`?

Reply:
I would rather not duck-type on a private attribute.  For example, it would seem a bit silly for the doc to say that the argument can be anything that has an _alias_map attribute.  If the docs does say "mapping or Artist subclass or Artist instance", then we may as well make the implementation match that.

Reply:
I don‘t see a problem with duck-typing on a private attribute if it‘s documented. That would be the low-overhead variant of a KeywordShortcutsMixin. It would allow to use it for other stuff than Artists.

However, we can always revisit later and lift that restriction.